test/Makefile

```diff
@@ -1,25 +1,50 @@
 DIR_MAKEFILE := ./make
+DIR_LOG      := ../log
 
-ifndef SYSTEM
-    ifeq ($(OS), Windows_NT)
-        SYSTEM := windows
-    else
-        SYSTEM := $(shell uname -s)
-
-        ifeq ($(filter $(SYSTEM),Linux Darwin),$(SYSTEM))
-            SYSTEM := unix_like
+ifeq (,$(wildcard $(DIR_LOG)/*.sys))
+    $(info There is no log including your operating system...)
+    ifndef SYSTEM
+        ifeq ($(OS), Windows_NT)
+            SYSTEM := windows
         else
-            SYSTEM := unknown
+            SYSTEM := $(shell uname -s)
+
+            ifeq ($(filter $(SYSTEM),Linux Darwin),$(SYSTEM))
+                SYSTEM := unix_like
+            else
+                SYSTEM := unknown
+            endif
+
         endif
+    endif
 
+    $(shell echo $(SYSTEM) > $(DIR_LOG)/$(SYSTEM).sys)
+    $(info Log including your operating system has been created!)
+else
+    $(info File .sys found!)
```

Comment:
Delete this msg. If it's found than everything is ok.
If you really want some msg to be displayed, change it to `Valid syslof file found!`

Reply:
Done.